client/signup/config/flows.js

```diff
@@ -55,7 +55,7 @@ function getPostsDestination( dependencies ) {
 function getRedirectDestination( dependencies ) {
 	if (
 		dependencies.oauth2_redirect &&
-		dependencies.oauth2_redirect.match( /^https%3A%2F%2Fpublic-api.wordpress.com/i )
+		dependencies.oauth2_redirect.match( /^https:\/\/public-api.wordpress.com/i )
```

Comment:
shouldn't the periods in the url be escaped with `\`?

Reply:
might be safer to use `.startsWith()` instead of match

Reply:
Updated to use this approach.